### PR TITLE
[REVIEW] Zero initialize gdf_column local variables. Fixes #1913

### LINKED ISSUE #1913: [BUG] cudf::apply_boolean_mask return value (gdf_column) has an invalid col_name pointer
**Describe the bug**
cudf::apply_boolean_mask return value (gdf_column) has an invalid col_name pointer (garbage value).

**Steps/Code to reproduce bug**

- Call cudf::apply_boolean_mask.
- Check that the col_name attribute of the returned gdf_column has a garbage value.

**Expected behavior**
cudf::apply_boolean_mask return value (gdf_column) col_name attribute should be a null pointer.

### PR DESCRIPTION
**Pull request by jeanp413**
This PR fixes issue #1913.
Additionally I looked at more places where the initialization was missing and updated them.

### CODE REVIEW

File: cpp/tests/stream_compaction/stream_compaction_tests.cu [hunk 1]
```diff
@@ -93,7 +93,7 @@ void BooleanMaskTest(cudf::test::column_wrapper<T> source,
                      cudf::test::column_wrapper<cudf::bool8> mask,
                      cudf::test::column_wrapper<T> expected)
 {
-  gdf_column result;
+  gdf_column result{};
   EXPECT_NO_THROW(result = cudf::apply_boolean_mask(source, mask));
 
   EXPECT_TRUE(expected == result);
```
Comment (harrism) [inline cpp/tests/stream_compaction/stream_compaction_tests.cu]: I think this line showcases where this PR could go further. Shouldn't `column_wrapper<T>::operator==()` also compare names? Then we would also make sure that `expected` contains a nullptr for name.

**Comment by jeanp413** [comment]
Regarding `gdf_column_view` and `gdf_column_view_extended`, should I add an additional parameter to both of them? Also not sure what should be the expected behavior:
- If the argument is nullptr, set col_name to nullptr
- Otherwise 
  - Copy the pointer value, or,
  - Copy the string using `strcpy` assuming col_name points to a large enough valid memory location

**Comment by harrism** [comment]
I'm not sure. I was just thinking you should have them initialize the `col_name` to `nullptr`, no extra parameters. Since we are redesigning the column this code will be temporary anyway.

I think eventually the `col_name` should be a `std::string` anyway.

**Comment by jeanp413** [comment]
As pointed out in https://github.com/rapidsai/cudf/issues/1904#issuecomment-498254378 setting the `col_name` to nullptr inside `gdf_column_view` and `gdf_column_view_extended` could be an unexpected side effect and code like this will need to be updated/avoided
https://github.com/rapidsai/cudf/blob/d2faef8cee4fd4e4a05d6c0bfd77b40b51c9b718/cpp/src/io/utilities/wrapper_utils.hpp#L38-L44

I think as log as we use `gdf_column col{};` and `new gdf_column col{};` would be enough

File: cpp/tests/stream_compaction/stream_compaction_tests.cu (hunk 1 above)
Comment (jeanp413) [reply cpp/tests/stream_compaction/stream_compaction_tests.cu]: Updated the `gdf_equal_columns` method in cudf_test_utils.cu

**Comment by harrism** [comment]
> think as long as we use `gdf_column col{};` and `new gdf_column col{};` would be enough

OK.

**Pull request merged**